incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/DumpHandlers.java

```diff
@@ -0,0 +1,255 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.command.dump.internal.airline;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.agrona.DirectBuffer;
+import org.pcap4j.core.NotOpenException;
+import org.pcap4j.core.PcapDumper;
+import org.pcap4j.core.PcapNativeException;
+import org.pcap4j.packet.EthernetPacket;
+import org.pcap4j.packet.IpV6Packet;
+import org.pcap4j.packet.IpV6SimpleFlowLabel;
+import org.pcap4j.packet.IpV6SimpleTrafficClass;
+import org.pcap4j.packet.Packet;
+import org.pcap4j.packet.TcpPacket;
+import org.pcap4j.packet.UnknownPacket;
+import org.pcap4j.packet.namednumber.EtherType;
+import org.pcap4j.packet.namednumber.IpNumber;
+import org.pcap4j.packet.namednumber.IpVersion;
+import org.pcap4j.packet.namednumber.TcpPort;
+import org.pcap4j.util.MacAddress;
+
+import com.google.common.primitives.Bytes;
+import com.google.common.primitives.Longs;
+import com.google.common.primitives.Shorts;
+
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.AbortFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.BeginFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.ChallengeFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.DataFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.EndFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.FlushFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.ResetFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.SignalFW;
+import io.aklivity.zilla.runtime.command.dump.internal.types.stream.WindowFW;
+
+public class DumpHandlers implements Handlers
+{
+    private final PcapDumper dumper;
+    private static final AtomicInteger SEQUENCE = new AtomicInteger(0);
+
+    public DumpHandlers(PcapDumper dumper)
+    {
+        this.dumper = dumper;
+    }
+    @Override
+    public void onBegin(BeginFW begin)
+    {
+        final long routeId = begin.routeId();
+        final long streamId = begin.streamId();
+        final int bindingId = (int) (routeId >> 0) & 0xffff_ffff;
+        Inet6Address address = createAddress(bindingId, streamId);
+        TcpPort port = getPort(streamId);
+        writePacketsToPcap(createPseudoEthernetPacket(createPseudoTcpPacketBuilder(address, port).syn(true),
```

Comment:
This will need to execute at very high performance, racing to keep up with the Zilla engine itself.

Therefore, while this approach is functionally correct, please try _not_ to create any objects during processing, as they would be created at a high rate, and trigger corresponding garbage collection effort.

Switching over to using generated `pcap` flyweights may help address this point as well as other issues using the `pcap4j` library.